streams/src/test/java/org/apache/kafka/streams/integration/QueryableStateIntegrationTest.java

```diff
@@ -361,7 +364,7 @@ public boolean conditionMet() {
                         final int index = metadata.hostInfo().port();
                         final KafkaStreams streamsWithKey = streamRunnables[index].getStream();
                         final ReadOnlyWindowStore<String, Long> store = streamsWithKey.store(storeName, QueryableStoreTypes.<String, Long>windowStore());
-                        return store != null && store.fetch(key, from, to) != null;
+                        return store != null && store.fetch(key, ofEpochMilli(from), ofMillis(to - from)) != null;
```

Comment:
This change indicate the change in behavior from "point-in-time/point-in-time" to "point-in-time/range" -- I cannot recall a discussion in the KIP about this change (did we discuss it?).

\cc @guozhangwang @bbejeck @vvcephei

Reply:
It was in the KIP proposal from the beginning, I think. I did notice it, and I think I mentioned it at some point as an example of how the new APIs could be superior, since it's much harder to mess up a call to this API than the "from/to" params, which could be accidentally inverted.

I don't think it ever came up as a proper discussion point, though.

For my part, I think that the choice between Instant/Duration and Instant/Instant is effectively a toss-up. I don't see it being a change in behavior, just a change in the expression.

I'm sure there are programs that would like it either way. If it's really a significant impediment to people, we can always tack on Instant/Instant later with a default implementation to defer to the other.

But you have been involved with the project for longer. Do you see "find me all the windows today between 0900 and 1000" as being more convenient than "find me all the windows within an hour after 0900 today" ?

Reply:
Guys, I'm waiting for your decision with this part of changes.
Please, write to me, if I need to make it `Instant, Instant`.

Reply:
For me, I have a slight preference for the `from` - `to` approach, but that's due more to personal preference.
I agree with @vvcephei that either way it's the same semantics just expressed differently.

So overall I'd prefer `Instant`, `Instant`, but if others want `Instant`, `Duration` I'm good with that as well.

Reply:
My main concern is different semantics for different methods: the old methods talking `long/long` have point-in-time/point-in-time semantics, while the new API has point-in-time/duration semantics. I would like to understand the impact of this change.

Especially, for `WindowedStore` we will now have both method (`long/long` is only deprecated on `ReadOnlyWindowedStore` but newly "added" to `WindowedStore`). It seems reasonable to me, to provide the same semantics for both -- overall, `long/long` should be a more performance runtime version of `Instant/Xxxx` and thus, offering different semantics using `Instant/Duration` (instead of `Instant/Instant`) might confuse users.

@nicktrav It seem to me, that we should discuss this carefully and maybe not rush into `2.1` release... Thoughts?

Reply:
Just synced up with the others -- seems we are in favor to switch to `Instant/Instant` and try to get it into `2.1` (see my other comment).